platform/src/main/java/platform/pageobjects/MyPage/ProductDetailsFilterValidation.java

```diff
@@ -0,0 +1,253 @@
+package platform.pageobjects.MyPage;
+
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.Select;
+
+import utils.CommonFunctions;
+import utils.ExcelUtil;
+
+/**
+ * @Author : Sopan Patil
+ * @Date : 12 Aug 2021
+ * @Description:Class for Product Details
+ */
+public class ProductDetailsFilterValidation {
+	List<String[]> testdata;
+	public WebDriver driver;
+	private static Logger log = LogManager.getLogger(ProductDetailsFilterValidation.class.getName());
+	int waitTime = 5;
+	String casecProduct, kanjikenteiProduct;
+	String freeuser, primeuser;
+
+	public ProductDetailsFilterValidation(WebDriver driver) {
+		this.driver = driver;
+		PageFactory.initElements(driver, this);
+	}
+
+	@FindBy(xpath = "//div[@class='header__menu d-md-block d-none']/ul/li[4]")
+	public WebElement productDetails;
+
+	@FindBy(xpath = "//a[@class='button button--blue1 button--xsmall font-weight-normal']/span")
+	public WebElement freeMemberTitle;
+
+	@FindBy(xpath = "//a[@class='button button--blue1 button--xsmall font-weight-normal']/span")
+	public WebElement primeMemberTitle;
+
+	@FindBy(xpath = "//select[@id='category']")
+	public WebElement productDropdown;
+
+	@FindBy(xpath = "//select[@id='type']")
+	public WebElement productTypeDropdown;
+
+	@FindBy(xpath = "//div[@class='ep-filter__button']")
+	public WebElement narrowDownButton;
+
+	@FindBy(xpath = "//div[@class='ep-product-list__row']")
+	public WebElement productListCard;
+
+	public void clickProductList() {
+		if (CommonFunctions.waitForVisiblity(productDetails, waitTime)) {
+			productDetails.click();
+		}
+	}
+
+	public void CheckingProductFilterFeature() throws Exception {
+
+		ExcelUtil excel = new ExcelUtil();
+		excel.setExcelFile("NewTestData.xlsx", "MembershipStatus");
+		freeuser = excel.getCellData("Freeuser", 1);
+		primeuser = excel.getCellData("Primeuser", 1);
+		if (freeMemberTitle.getText().contentEquals(freeuser)) {
+			productDetails.click();
+			log.info("This is free user");
+			CheckFreeMembership_CASEC_ProductDetails();
+			CheckFreeMembershipKanjikenteiProductDetails();
+		} else if (primeMemberTitle.getText().contentEquals(primeuser)) {
+			productDetails.click();
+			log.info("This is prime user");
+			CheckPrimeMembership_CASEC_Productetails();
+			CheckPrimeMembership_kanjikentei_Productetails();
+		}
+	}
+
+	public void CheckFreeMembership_CASEC_ProductDetails() throws Exception {
```

Comment:
Please remove _ (underscore) and spell chk

Reply:
Done